alkira/resource_alkira_connector_akamai_prolexic.go

```diff
@@ -0,0 +1,256 @@
+				Optional:    true,
+			},
+			"size": &schema.Schema{
+				Description:  "The size of the connector. one of `SMALL`, `MEDIUM` and `LARGE`.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"SMALL", "MEDIUM", "LARGE"}, false),
+			},
+			"segment_id": {
+				Description: "The ID of segments associated with the connector. Currently, only `1` segment is allowed.",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
+			"tunnel_configuration": &schema.Schema{
+				Description: "Tunnel Configurations.",
+				Type:        schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alkira_public_ip": {
+							Description: "Alkira public IP.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"tunnel_ips": {
+							Description: "Tunnel IPs.",
+							Type:        schema.TypeSet,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"ran_tunnel_ip": {
+										Description: "The underlay tunnel IP on the Akamai " +
+											"side to be used to configure tunnels between " +
+											"the Alkira CXP and the Akamai Prolexic service. " +
+											"A RAN (Routed Access Network) is the unit of " +
+											"availability for the Route GRE 3.0 service.",
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"alkira_overlay_tunnel_ip": {
+										Description: "The overlay IP of the GRE tunnel on the Alkira side.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"akamai_overlay_tunnel_ip": {
+										Description: "The overlay IP of the GRE tunnel on the Alkira side.",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+								},
+							},
+							Required: true,
+						},
+					},
+				},
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceConnectorAkamaiProlexicCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+	connector, err := generateConnectorAkamaiProlexicRequest(client, d, m)
+
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Creating Connector (Akamai-Prolexic)")
+	id, err := client.CreateConnectorAkamaiProlexic(connector)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(id)
+
+	return resourceConnectorAkamaiProlexicRead(d, m)
+}
+
+func resourceConnectorAkamaiProlexicRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+
+	connector, err := client.GetConnectorAkamaiProlexic(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("akamai_bgp_asn", connector.AkamaiBgpAsn)
+	d.Set("billing_tag_ids", connector.BillingTags)
+	d.Set("cxp", connector.CXP)
+	d.Set("group", connector.Group)
+	d.Set("enabled", connector.Enabled)
+	d.Set("name", connector.Name)
+	d.Set("size", connector.Size)
+
+	if len(connector.Segments) > 0 {
```

Comment:
This check should be placed nearer the top of the function. No need to set all those other fields if the check for connector segment fails. Not a big deal here but a solid practice nonetheless.

Reply:
No, even segment is not valid. We still want to set the fields as much as possible, eventually, it will produce a diff.. The whole point of read is to generate a diff as much as we could.